src/ClientTransactionTask.cc

```diff
@@ -0,0 +1,562 @@
+        status = e.status;
+        state = DONE;
+    }
+}
+
+/**
+ * Initialize all necessary values of the commit task in preparation for the
+ * commit protocol.  This includes building the send-ready buffer of
+ * participants to be included in every prepare rpc and also the allocation of
+ * rpcIds.  Used in the commit method.  Factored out mostly for ease of testing.
+ */
+void
+ClientTransactionTask::initTask()
+{
+    lease = ramcloud->clientLease.getLease();
+    txId = ramcloud->rpcTracker.newRpcId(1);    // TODO(cstlee) : see todo in
+                                                // the method call.
+
+    nextCacheEntry = commitCache.begin();
+    while (nextCacheEntry != commitCache.end()) {
+        const CacheKey* key = &nextCacheEntry->first;
+        CacheEntry* entry = &nextCacheEntry->second;
+
+        entry->rpcId = ramcloud->rpcTracker.newRpcId(1);    // TODO(cstlee) :
+                                                            // see todo in the
+                                                            // method call.
+        participantList.emplaceAppend<WireFormat::TxParticipant>(
+                key->tableId,
+                static_cast<uint64_t>(key->keyHash),
+                entry->rpcId);
+        participantCount++;
+        nextCacheEntry++;
+    }
+}
+
+/**
+ * Process any decision rpcs that have completed.  Used in the commit method.
+ * Factored out mostly for ease of testing.
+ */
+void
+ClientTransactionTask::processDecisionRpcs()
+{
+    // Process outstanding RPCs.
+    std::list<DecisionRpc>::iterator it = decisionRpcs.begin();
+    for (; it != decisionRpcs.end(); it++) {
+        DecisionRpc* rpc = &(*it);
+
+        if (!rpc->isReady()) {
+            continue;
+        }
+
+        if (rpc->getState() == rpc->FAILED) {
+            // Nothing to do.  Will be retried.
+            TEST_LOG("FAILED");
+        } else if (rpc->responseHeader->status == STATUS_OK) {
+            TEST_LOG("STATUS_OK");
+            for (uint32_t i = 0; i < rpc->reqHdr->participantCount; i++)
+                ramcloud->rpcTracker.rpcFinished(rpc->ops[i]->second.rpcId);
+        } else if (rpc->responseHeader->status == STATUS_UNKNOWN_TABLET) {
+            // Nothing to do.  Will be retried.
+            TEST_LOG("STATUS_UNKNOWN_TABLET");
+        } else {
+            ClientException::throwException(HERE, rpc->responseHeader->status);
+        }
+
+        // Destroy object.
+        it = decisionRpcs.erase(it);
+    }
+}
+
+/**
+ * Process any prepare rpcs that have completed.  Used in the commit method.
+ * Factored out mostly for ease of testing.
+ */
+void
+ClientTransactionTask::processPrepareRpcs()
+{
+    // Process outstanding RPCs.
+    std::list<PrepareRpc>::iterator it = prepareRpcs.begin();
+    for (; it != prepareRpcs.end(); it++) {
+        PrepareRpc* rpc = &(*it);
+
+        if (!rpc->isReady()) {
+            continue;
+        }
+
+        if (rpc->getState() == rpc->FAILED) {
```

Comment:
"rpc->FAILED" is strange notation; even if it works (which it seems to do), wouldn't it be clearer to spell this out as RpcWrapper::RpcState::FAILED?